llvm/lib/Target/AMDGPU/SIISelLowering.cpp

```diff
@@ -13857,6 +13857,52 @@ static SDValue getMad64_32(SelectionDAG &DAG, const SDLoc &SL, EVT VT,
   return DAG.getNode(ISD::TRUNCATE, SL, VT, Mad);
 }
 
+// Fold
+//     y = lshr i64 x, 32
+//     res = add (mul i64 y, Constant), x   where "Constant" is a 32 bit
+//     negative value
```

Comment:
"32 bit negative value" does not describe it properly. It's a 64-bit constant where the high 32 bits are all-ones. Or using the notation below you could say `Constant.hi == -1`

Reply:
comment updated